openlibrary/plugins/openlibrary/js/check-ins/index.js

```diff
@@ -0,0 +1,348 @@
+ */
+function submitEvent(elem) {
+    const eventType = Number(elem.dataset.eventType)
+    const url = elem.querySelector('form').action
+
+    const editField = elem.querySelector('input[name=is_edit]')
+    const isEdit = editField.value === 'true'
+
+    const editionField = elem.querySelector('input[name=edition_key]')
+    const editionKey = editionField ? editionField.value : null
+
+    const dayField = elem.querySelector('select[name=day]')
+    const day = dayField.value
+    const monthField = elem.querySelector('select[name=month]')
+    const month = monthField.value
+    const yearField = elem.querySelector('select[name=year]')
+    const year = yearField.value
+
+    const data = {
+        event_type: Number(eventType),
+        year: year ? Number(year) : null,
+        month: month ? Number(month) : null,
+        day: day ? Number(day) : null,
+        is_edit: isEdit
+    }
+
+    if (editionKey) {
+        data.edition_key = editionKey
+    }
+
+    $.ajax({
+        type: 'POST',
+        url: url,
+        contentType: 'application/json',
+        data: JSON.stringify(data),
+        dataType: 'json',
+        beforeSend: function(xhr) {
+            xhr.setRequestHeader('Content-Type', 'application/json');
+            xhr.setRequestHeader('Accept', 'application/json');
+        },
+        success: function() {
+            const editField = elem.querySelector('input[name=is_edit]')
+            editField.value = true
+            updateView(elem.dataset.workOlid, year, month, day)
+        },
+        error: function() {
+            new PersistentToast('Failed to submit check-in.  Please try again in a few moments.').show()
+        },
+        complete: function() {
+            closeDialog(elem.dataset.workOlid)
+        }
+    });
+}
+
+/**
+ * Dispatches close dialog event to the parent dialog element.
+ *
+ * @param {string} workOlid Uniquely identifies a check-in dialog element.
+ */
+function closeDialog(workOlid) {
+    const dialog = document.querySelector(`#check-in-dialog-${workOlid}`)
+    dialog.dispatchEvent(new Event('close-dialog'))
+}
+
+/**
+ * Updates and displays check-in date.
+ *
+ * Removes check-in prompt component if it exists.
+ *
+ * @param {string} workOlid ID used to identify related components
+ * @param {str} year Check-in event year
+ * @param {str|null} month Check-in event month
+ * @param {str|null} day Check-in event day
+ */
+function updateView(workOlid, year, month, day) {
+    let date = year
+    if (month) {
+        date += `-${month}`
+        if (day) {
+            date += `-${day}`
+        }
+    }
+    const displayElement = document.querySelector(`#check-in-display-${workOlid}`)
+    const dateField = displayElement.querySelector('.check-in-date')
+    dateField.textContent = date
+
+    const promptElem = document.querySelector(`#prompt-${workOlid}`)
+    if (promptElem) {
+        promptElem.remove()
```

Comment:
```suggestion
        promptElem.classList.add('hidden')
```
Should not remove, as the "add a last read date" prompt will be needed once deletion on books page is possible.